### PR TITLE
[CLEANUP] Update the configuration proxy to the new configuration format

### LINKED ISSUE #465: Update the configuration proxy to the new configuration format
https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/9.0/Deprecation-82254-DeprecateGLOBALSTYPO3_CONF_VARSEXTextConf.html

We need to support both, and if both are present, the new format needs to have a higher priority.

Mark the old way as `@deprecated` and to be removed in oelib 4.0.

Part of #419.

### PR DESCRIPTION
**Pull request by Konafets**
Closes #465

### CODE REVIEW

**Comment by Konafets** [comment]
How would you add a `@deprecate` flag to the old way as you request in the issue?

File: Classes/Templating/TemplateHelper.php [hunk 1]
```diff
@@ -176,17 +176,7 @@ protected function getConfigurationCheckClassName()
...
-        if (\class_exists($camelCaseClassName)) {
-            $className = $camelCaseClassName;
-        } elseif (\class_exists($lowercaseClassName)) {
-            $className = $lowercaseClassName;
-        } else {
-            $className = '';
-        }
-
-        return $className;
+        return ConfigurationCheck::class;
```
Comment (oliverklee) [inline Classes/Templating/TemplateHelper.php:179]: Please undo the changes to this file. They are not related to the issue to be solved, and they are required for configuration check to work on extensions like seminars 3.x that still rely on the configuration check by class name.

**Review by oliverklee — CHANGES_REQUESTED** [review]

File: Classes/Configuration/ConfigurationProxy.php [hunk 2]
```diff
@@ -185,4 +190,14 @@ public function getCompleteConfiguration(): array
 
         return $this->configuration;
     }
+
+    private function hasNewConfigurationFormat(): bool
```
Comment (oliverklee) [inline Classes/Configuration/ConfigurationProxy.php:194]: Please also mark both functions as deprecated and to be removed in oelib 4.0 as we'll drop support for the old way after dropping support for TYPO3 8.7.

**Comment by oliverklee** [comment]
> How would you add a `@deprecate` flag to the old way as you request in the issue?

We can deprecate the new methods (and add a comment that only the new way will be supported in oelib 4.0).

**Comment by oliverklee** [comment]
This also needs a changelog entry.

**Review by oliverklee — APPROVED** [review]

**Pull request merged**